ui/stories/components/table.stories.js

```diff
@@ -0,0 +1,483 @@
+import hbs from 'htmlbars-inline-precompile';
+import productMetadata from '../../app/utils/styleguide/product-metadata';
+
+import EmberObject, { computed } from '@ember/object';
+
+import { getOwner } from '@ember/application';
+import { on } from '@ember/object/evented';
+import Controller from '@ember/controller';
+
+export default {
+  title: 'Components|Table',
+};
+
+/**
+ * The Ember integration for Storybook renders a container component with no routing,
+ * which means things that need query parameters, like sorting and pagination, won’t work.
+
+ * This initialiser turns on routing and accepts a controller definition that gets wired up
+ * to a generated `storybook` route. The controller is attached to the Storybook component
+ * as the `controller` property so its query parameters are accessible from the template.
+ */
+function injectRoutedController(controllerClass) {
```

Comment:
Is this (injectRoutedController) something that would be worthwhile externalizing and importing here?

Reply:
Since this is the only component that needs routing, it didn’t seem worth it to me, à la “rule of three”, but I could if it’s desired.